AutomatedTesting/Gem/PythonTests/Atom/atom_utils/material_canvas_utils.py

```diff
@@ -8,103 +8,142 @@
-    """
-    return graph.GraphControllerRequestBus(
-        bus.Event, GraphControllerRequestBusEvents.ADD_NODE, graph_id, node, position)
-
-
-def get_graph_model_slot_id(slot_name: str) -> object:
-    """
-    Given a slot_name string, return a GraphModelSlotId object representing a node slot.
-    :param slot_name: String representing the name of the slot to target on the node.
-    :return: An GraphModelSlotId object.
-    """
-    return graph.GraphModelSlotId(slot_name)
-
-
-def add_connection_by_slot_id(
-        graph_id: math.Uuid,
-        source_node: object, source_slot: object,
-        target_node: object, target_slot: object) -> object:
-    """
-    Adds a new connection between a source node slot and a target node slot.
-    :param graph_id: int representing the ID value of a given graph AZStd::shared_ptr<Graph> object.
-    :param source_node: A proxy AZStd::shared_ptr<Node> object for the source node to start the connection from.
-    :param source_slot: A proxy GraphModelSlotId object for the slot on the source node to use for the connection.
-    :param target_node: A proxy AZStd::shared_ptr<Node> object for the target node to end the connection to.
-    :param target_slot: A proxy GraphModelSlotId object for the slot on the target node to use for the connection.
-    :return: azlmbr.object.PythonProxyObject representing a C++ AZStd::shared_ptr<Connection> object.
-    """
-    return graph.GraphControllerRequestBus(
-        bus.Event, GraphControllerRequestBusEvents.ADD_CONNECTION_BY_SLOT_ID, graph_id,
-        source_node, source_slot, target_node, target_slot)
-
-
-def are_slots_connected(
-        graph_id: math.Uuid,
-        source_node: object, source_slot: object,
-        target_node: object, target_slot: object) -> bool:
-    """
-    Determines if 2 nodes have a connection formed between them and returns a boolean for success/failure.
-    :param graph_id: int representing the ID value of a given graph AZStd::shared_ptr<Graph> object.
-    :param source_node: An AZStd::shared_ptr<Node> object representing the source node for the connection.
-    :param source_slot: An GraphModelSlotId object representing the slot on the source node the connection uses.
-    :param target_node: An AZStd::shared_ptr<Node> object representing the target node for the connection.
-    :param target_slot: An GraphModelSlotId object representing the slot on the target node the connection uses.
-    :return: bool representing success (True) or failure (False).
-    """
-    return graph.GraphControllerRequestBus(
-        bus.Event, GraphControllerRequestBusEvents.ARE_SLOTS_CONNECTED, graph_id,
-        source_node, source_slot, target_node, target_slot)
+class Node(object):
+    """
+    Represents a node inside of a graph and contains node related functions.
+    """
+
+    def __init__(self, node_python_proxy: azlmbr.object.PythonProxyObject, node_id: azlmbr.object.PythonProxyObject):
```

Comment:
you don't need to pass in the node id because you can get it from the node
it might be safest to only pass the document id and node id instead then always lookup the graph and node
either way